### PR TITLE
Detect stream frames/v20

### PR DESCRIPTION
**Pull request by victorjulien**
Stream and frame updates. Draft for CI/QA.

suricata-verify-pr: 1067

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_single_stats_chk | | |
| .tcp.ssn_from_cache | 264738 | 340152 | 128.49% |
| .tcp.ssn_from_pool | 628253 | 552839 | 88.0% |
| .tcp.segment_from_pool | 2384586 | 2542767 | 106.63% |
| .tcp.memuse | 88800000 | 93600000 | 105.41% |
| | SURI_TLPW2_autofp_stats_chk | | |
| .tcp.ssn_from_cache | 298309 | 383592 | 128.59% |
| .tcp.ssn_from_pool | 594682 | 509399 | 85.66% |
| .tcp.segment_from_pool | 2040566 | 2250999 | 110.31% |
| .tcp.memuse | 5683200000 | 5990400000 | 105.41% |
| | SURI_TLPW1_stats_chk | | |
| .tcp.ssn_from_cache | 63648 | 79238 | 124.49% |
| .tcp.ssn_from_pool | 139168 | 123578 | 88.8% |
| .tcp.rst | 128237 | 145039 | 113.1% |
| .tcp.segment_from_pool | 251657 | 404269 | 160.64% |
| .tcp.memuse | 88800000 | 93600000 | 105.41% |
| .app_layer.error.tls.parser | 931 | 1145 | 122.99% |
| | SURI_TLPR1_stats_chk | | |
| .flow.memuse | 565135168 | 519185408 | 91.87% |
| .tcp.ssn_from_cache | 3647103 | 5744106 | 157.5% |
| .tcp.ssn_from_pool | 8482654 | 6385657 | 75.28% |
| .tcp.synack | 7487314 | 10198881 | 136.22% |
| .tcp.rst | 5307555 | 6695438 | 126.15% |
| .tcp.segment_from_pool | 8600661 | 9443931 | 109.8% |
| .tcp.memuse | 5683200000 | 5990400000 | 105.41% |
| .app_layer.flow.http | 3118951 | 3465820 | 111.12% |
| .app_layer.flow.tls | 510462 | 624265 | 122.29% |
| .app_layer.flow.ssh | 8937 | 12678 | 141.86% |
| .app_layer.flow.telnet | 1624 | 2358 | 145.2% |
| | IPS_AFP_stats_chk | | |
| .tcp.memuse | 1776000000 | 1872000000 | 105.41% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 895832 | 0.15 | 

Pipeline 11711

File: src/detect-engine-frame.c [hunk 1]
```diff
@@ -200,77 +204,99 @@ struct FrameStreamData {
...
+            if (so_input_re >= so_frame_re) {
                 ci_flags |= DETECT_CI_FLAGS_END;
                 SCLogDebug("have frame data end");
+                in_data_excess = so_input_re - so_frame_re;
             }
+            data_len = input_len - in_data_offset - in_data_excess;
         } else {
-            data_len = input_len;
+            /* in: relative to start of input data */
+            BUG_ON(so_frame_inspect_offset < offset);
```
Comment (victorjulien) [inline src/detect-engine-frame.c:286]: may happen after a gap?

File: src/stream-tcp-reassemble.c [hunk 2]
```diff
@@ -484,7 +484,23 @@ static int StreamTcpReassemblyConfig(bool quiet)
         StreamTcpReassembleConfigEnableOverlapCheck();
     }
 
+    uint16_t max_regions = 8;
```
Comment (victorjulien) [inline src/stream-tcp-reassemble.c:487]: use macro

**Comment by victorjulien** [comment]
replaced by #8405

**Pull request closed without merging**